assets/src/components/appStateWrapper.tsx

```diff
@@ -4,19 +4,31 @@ import useSocket from "../hooks/useSocket"
 import useTimepoints from "../hooks/useTimepoints"
 import useVehicles from "../hooks/useVehicles"
 import DispatchProvider from "../providers/dispatchProvider"
-import { Route, TimepointsByRouteId } from "../skate"
+import { Route, TimepointsByRouteId, UserToken } from "../skate"
 import { initialState, reducer } from "../state"
 import App from "./app"
 
+export const readUserToken = (): UserToken | undefined => {
+  const dataEl = document.getElementById("app")
+  if (!dataEl) {
+    return undefined
+  }
+
+  const token = dataEl.dataset.userToken as UserToken
+  return token
+}
+const userToken: UserToken | undefined = readUserToken()
```

Comment:
We could, though that would mean running this logic on each network request, instead of once when the app loads. Does that tradeoff seem worthwhile? I feel like my first instinct is to lean towards no, but I don't feel strongly.

Oh, we'd also need to do it in the socket as well.

Reply:
Ah, yeah, we should only read it once. 👍